webpack/assets/javascripts/react_app/constants.js

```diff
@@ -3,3 +3,22 @@ export const STATUS = {
   RESOLVED: 'RESOLVED',
   ERROR: 'ERROR',
 };
+
+export const getControllerSearchProps = (
+  controller,
+  id = 'searchBar',
+  canCreate = true
+) => ({
+  controller,
+  autocomplete: {
+    id,
+    searchQuery: '',
+    url: `${controller}/auto_complete_search`,
+    useKeyShortcuts: true,
+  },
+  bookmarks: {
+    url: '/api/bookmarks',
+    canCreate,
+    documentationUrl: `4.1.5Searching`,
```

Comment:
again, no props from the server. thats why its now in redux

Reply:
in the last comments I meant using / implementing those methods that are used in the server.
Receiving them as `metadata` is also optional, isn't it ?

Reply:
instead of maintaining the functions in two places we receive it [here](https://github.com/theforeman/foreman/pull/6767/files#diff-127a850b7d7b20173ad3fba88f8a25e3R532) for now.
could be good to move the implementation to the client later